test/acceptance/features/examples.feature

```diff
@@ -53,3 +54,54 @@ Feature: Verify examples provided in Service Binding Operator github repository
+        When Service Binding is applied
+            """
+            apiVersion: operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: sbr-bind-mul-svc-ex
+            spec:
+                application:
+                    name: node-todo-mul-svc-ex
+                    group: apps
+                    version: v1
+                    resource: deployments
+                services:
+                - group: postgresql.baiju.dev
+                  version: v1alpha1
+                  kind: Database
+                  name: db-mul-svc-ex
+                - group: etcd.database.coreos.com
+                  version: v1beta2
+                  kind: EtcdCluster
+                  name: etcd-mul-svc-ex
+                detectBindingResources: true
+            """
+        Then Service Binding "sbr-bind-mul-svc-ex" is ready
+        And Secret "sbr-bind-mul-svc-ex" contains "DATABASE_DBNAME" key with value "db-mul-svc-ex"
+        And Secret "sbr-bind-mul-svc-ex" contains "DATABASE_USER" key with value "postgres"
+        And Secret "sbr-bind-mul-svc-ex" contains "DATABASE_PASSWORD" key with value "password"
+        And Secret "sbr-bind-mul-svc-ex" contains "DATABASE_DB_PASSWORD" key with value "password"
+        And Secret "sbr-bind-mul-svc-ex" contains "DATABASE_DB_NAME" key with value "db-mul-svc-ex"
+        And Secret "sbr-bind-mul-svc-ex" contains "DATABASE_DB_PORT" key with value "5432"
+        And Secret "sbr-bind-mul-svc-ex" contains "DATABASE_DB_USER" key with value "postgres"
+        And Secret "sbr-bind-mul-svc-ex" contains "DATABASE_DB_HOST" key with dynamic IP addess as the value
+        And Secret "sbr-bind-mul-svc-ex" contains "DATABASE_DBCONNECTIONIP" key with dynamic IP addess as the value
+        And Secret "sbr-bind-mul-svc-ex" contains "DATABASE_DBCONNECTIONPORT" key with value "5432"
+        And Secret "sbr-bind-mul-svc-ex" contains "DATABASE_IMAGE" key with value "docker.io/postgres"
+        And Secret "sbr-bind-mul-svc-ex" contains "DATABASE_IMAGENAME" key with value "postgres"
+        And Secret "sbr-bind-mul-svc-ex" contains "ETCDCLUSTER_CLUSTERIP" key with dynamic IP addess as the value
+        And Secret "sbr-bind-mul-svc-ex" contains "ETCDCLUSTER_DB_HOST" key with dynamic IP addess as the value
+        And Secret "sbr-bind-mul-svc-ex" contains "ETCDCLUSTER_DB_NAME" key with value "db-mul-svc-ex"
+        And Secret "sbr-bind-mul-svc-ex" contains "ETCDCLUSTER_DB_PASSWORD" key with value "password"
+        And Secret "sbr-bind-mul-svc-ex" contains "ETCDCLUSTER_DB_PORT" key with value "5432"
+        And Secret "sbr-bind-mul-svc-ex" contains "ETCDCLUSTER_DB_USER" key with value "postgres"
+        And Secret "sbr-bind-mul-svc-ex" contains "ETCDCLUSTER_PASSWORD" key with value "cGFzc3dvcmQ="
+        And Secret "sbr-bind-mul-svc-ex" contains "ETCDCLUSTER_USER" key with value "cG9zdGdyZXM="
```

Comment:
Do we have to? Because If I'm not wrong our primary purpose is to make sure secrets has the relevant data and it is made available to be consumed by the user or by the application. Now how that user is using it or how the application is consuming it is up to on their side. So making sure that secret is available and it has relevant data should be sufficient check on our side.

Reply:
This is the application used in demoing example https://github.com/akashshinde/node-todo. I don't see it uses etcd-cluster in any way, they just used it superficially. So at the application level, I don't see any way to make sure that service has been used by the app and it is working correctly.

Reply:
> Because If I'm not wrong our primary purpose is to make sure secrets has the relevant data and it is made available to be consumed by the user or by the application. Now how that user is using it or how the application is consuming it is up to on their side.

Usage of a secret to inject binding data is just an implementation detail, the operator could do that in a different way as well. The fact that the secret contains some data is not enough - because we could have a bug in the code, and not refer the secret at the end in the application deployment and hence, the application will not see the binding data.

This brings us to the example we want to use. IMHO, the example is not very useful, from the perspective of user and would need to be rewritten to either:

* use some real-world application that consumes two services, or
* we use the generic test application from the acceptance tests

I would prefer former. For idea: https://github.com/spring-petclinic/spring-petclinic-microservices

@pmacik wdyt?

Reply:
Yes, the example itself is rather academic, confusing and does not make much sense the way it is. We should modify it to have an application that actually consumes those multiple services - whatever that is...

Reply:
But ideally, it should be something real-world so that a user who wants to try&learn our example sees a real-world usage.